### DISCUSSION TITLE
Suggestion: Redirect issues with multiple copies of a site on the same server

### DISCUSSION THREAD

**Discussion by johfive** [rhubarbgroup/.github#55] [Support (ROC)] [answered]
I have an environment with multiple copies of a site (staging, live) on the same server.

I followed all the setup instructions, including setting wp-config constants for the Redis key prefixes and database numbers.

Despite this, when multiple sites had Redis caching active, I had issues with the sites redirecting to each other when browsing or logging in to Wordpress.

It turned out that the Wordpress salts (https://kinsta.com/knowledgebase/wordpress-salts/) were the same on all sites.
This is discouraged, but all sites worked fine until the Redis plugin was activated.
It's simple to fix - simply regenerate the salts so all sites are different, then flush all the Redis and browser caches.

I appreciate this might not be a bug in the plugin, more a Wordpress config thing -- but is perhaps worth mentioning in the setup instructions and/or FAQs? Browsing the web for reports of the redirects issue I found plenty, not all resolved. This particular issue could be a rare one but not easy to figure out.

Thanks :)

**Comment by tillkruss** [comment, marked as answer]
After setting the `WP_REDIS_PREFIX` for each site you must flush the Redis Server if you it already contains contaminated data.

**Comment by johfive** [comment]
Thanks :) I did that. 
So in summary:
1. Ensure Wordpress Salts are different per site
2. Ensure WP_REDIS_PREFIX and/or WP_REDIS_DATABASE are set, and different per site
3. Flush all Redis Caches
